code/modules/events/immovable_rod.dm

```diff
@@ -81,4 +83,13 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 			H.adjustBruteLoss(160)
 		if(clong.density || prob(10))
 			clong.ex_act(2)
-	return
\ No newline at end of file
+
+/obj/effect/immovablerod/event
+	var/tiles_moved = 0
+
+/obj/effect/immovablerod/event/Move()
+	var/atom/oldloc = loc
+	. = ..()
+	tiles_moved++
+	if(get_dist(oldloc, loc) > 2 && tiles_moved > 10) // We went on a journey, commit sudoku
```

Comment:
No, this means that if it ever jumps more than 2 tiles, and has already moved at least 10 tiles, it'll delete itself (this is symptomatic of crossing a z boundary)

This is necessary because in the event, the rod may cross the border once before actually approaching the station, which would delete the rod immediately if we didn't have the 10 tile lower bound. This way, the rod will have a guaranteed single pass without destroying itself before anything happens or claiming a column of the station.